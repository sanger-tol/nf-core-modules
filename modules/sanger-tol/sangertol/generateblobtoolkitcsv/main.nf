process SANGERTOL_GENERATEBLOBTOOLKITCSV {
    tag "${meta.id}"
    executor 'local'

    input:
    tuple val(meta), path(input_files)
    val(longread_technology)
    val(library_layout)

    output:
    tuple val(meta),    path("${prefix}.samplesheet.csv")   , emit: csv
    path("versions.yml")                                    , emit: versions

    when:
    task.ext.when == null || task.ext.when

    exec:
    // Note: Manually bump version number when updating module
    def VERSION     = "1.0.0"
    prefix          = task.ext.prefix ?: "${meta.id}"

    def fasta_files = input_files.findAll { it.name.endsWith('.fasta.gz') || it.name.endsWith('.fa.gz') }

    assert fasta_files.size() > 0
    def samplesheet_entries = ["sample,datatype,datafile,library_layout"]
    fasta_files.withIndex().each{ fa, idx -> samplesheet_entries << "${meta.id}_T${idx+1},${longread_technology},${fa},${library_layout}" }
    file(task.workDir.resolve("${prefix}.samplesheet.csv")).text = samplesheet_entries.join("\n")

    file("${task.workDir}/versions.yml").text = """\
        SANGERTOL_GENERATEBLOBTOOLKITCSV:
            sangertol_generateblobtoolkitcsv: ${VERSION}
        """.stripIndent()

}
