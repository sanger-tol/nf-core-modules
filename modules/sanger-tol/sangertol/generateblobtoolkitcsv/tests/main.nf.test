nextflow_process {

    name "Test Process SANGERTOL_GENERATEBLOBTOOLKITCSV"
    script "../main.nf"
    process "SANGERTOL_GENERATEBLOBTOOLKITCSV"

    tag "modules"
    tag "modules_sangertol"
    tag "sangertol"
    tag "sangertol/generateblobtoolkitcsv"

    test("fasta.gz to csv") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test' ], // meta map
                    [
                        file(params.modules_testdata_base_path + 'resources/ascc/asccTinyTest_V2/pacbio/Pyoeliiyoelii17XNL_pbtools_simulated_pacbio_reads.fa.gz', checkIfExists: true),
                        file(params.modules_testdata_base_path + 'resources/treeval/TreeValTinyData/genomic_data/pacbio/seqkitPacbio50000.fasta.gz', checkIfExists: true)
                    ]
                ]
                input[1] = "pacbio"
                input[2] = "SINGLE"
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
