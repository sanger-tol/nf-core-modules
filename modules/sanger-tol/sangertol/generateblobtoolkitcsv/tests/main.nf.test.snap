{
    "fasta.gz to csv": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.samplesheet.csv:md5,85f28ee33669df8cbe5c945988879e54"
                    ]
                ],
                "1": [
                    "versions.yml:md5,d4f404227c9dfc0dfd7d27fbf1e33557"
                ],
                "csv": [
                    [
                        {
                            "id": "test"
                        },
                        "test.samplesheet.csv:md5,85f28ee33669df8cbe5c945988879e54"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,d4f404227c9dfc0dfd7d27fbf1e33557"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-10-30T16:53:42.356305559"
    }
}